### PR TITLE
sanitize the solve_triu and sync with AA

### PR DESCRIPTION
**Pull request by fieker**
technically breaking as _solve_triu is now AA._solve_triu

### CODE REVIEW

File: src/flint/fmpz_mat.jl [hunk 1]
```diff
@@ -1554,7 +1554,11 @@ function _solve_dixon(a::ZZMatrix, b::ZZMatrix)
 end
 
 #XU = B. only the upper triangular part of U is used
-function _solve_triu_left(U::ZZMatrix, b::ZZMatrix)
+function AbstractAlgebra._solve_triu_left(U::ZZMatrix, b::ZZMatrix; side::Symbol = :left)
```
Comment (fingolfin) [inline src/flint/fmpz_mat.jl:1557]: There is no `AbstractAlgebra._solve_triu_left`
```suggestion
function _solve_triu_left(U::ZZMatrix, b::ZZMatrix; side::Symbol = :left)
```

File: src/flint/fmpz_mat.jl [hunk 2]
```diff
@@ -1554,7 +1554,11 @@ function _solve_dixon(a::ZZMatrix, b::ZZMatrix)
...
+  if side == :right
+    return AbstractAlgebra._solve_triu(U, b; side)
+  end
```
Comment (fingolfin) [inline src/flint/fmpz_mat.jl:1560]: uh... why does `_solve_triu_left` have a `side` parameter to begin with?!?

As far as I can tell *nothing* calls `_solve_triu_left` except one test case in Nemo.jl. So we could also just remove it?

File: src/flint/fmpz_mat.jl [hunk 3]
```diff
@@ -1595,8 +1599,12 @@ function _solve_triu_left(U::ZZMatrix, b::ZZMatrix)
   return X
 end
 
-#UX = B
-function _solve_triu(U::ZZMatrix, b::ZZMatrix) 
+#UX = B, U has to be upper triangular
+function AbstractAlgebra._solve_triu(U::ZZMatrix, b::ZZMatrix; side::Symbol=:left) 
+  if side == :left
+    return AbstractAlgebra._solve_triu_left(U, b; side)
```
Comment (fingolfin) [inline src/flint/fmpz_mat.jl:1605]:
```suggestion
    return _solve_triu_left(U, b; side)
```

File: src/flint/fmpz_mat.jl (hunk 2 above)
Comment (fieker) [reply src/flint/fmpz_mat.jl:1560]: No. solve_triu left is essential for the stuff in Hecke. The solve right is not used (currently)

Comment (lgoettgens) [reply src/flint/fmpz_mat.jl:1560]: I agree with @fingolfin that `_solve_triu` should have the `side` kwarg and dispatch to different methods depending on it. `_solve_triu_left` should not have to deal with these things.

**Pull request merged**